build.ps1

```diff
@@ -0,0 +1,109 @@
+<#
+.Synopsis 
+    Acquires dependencies, builds, and tests this project.
+.Description
+    If no actions are specified, the default is to run all actions.
+.Parameter Restore
+    Restore NuGet packages.
+.Parameter Build
+    Build the entire project. Requires that -Restore is or has been executed. 
+.Parameter Test
+    Run all built tests.
+.Parameter Configuration
+    The configuration to build. Either "debug" or "release". The default is debug.
+.Parameter WarnAsError
+    Converts all build warnings to errors. Useful in preparation to sending a pull request.
+#>
+[CmdletBinding(SupportsShouldProcess=$true,ConfirmImpact='Medium')]
+Param(
+    [switch]$Restore,
+    [switch]$Build,
+    [switch]$Test,
+    [Parameter()][ValidateSet('debug', 'release')]
+    [string]$Configuration = 'debug',
+    [switch]$WarnAsError = $true
+)
+
+$NothingToDo = !($Restore -or $Build -or $Test)
+if ($NothingToDo) {
+    Write-Output "No actions specified. Applying default actions."
+    $Restore = $true
+    $Build = $true
+    $Test = $true    
+}
+
+# External dependencies
+$sourceNugetExe = "https://dist.nuget.org/win-x86-commandline/v3.3.0/nuget.exe"
+
+# Path variables
+$ProjectRoot = Split-Path -parent $PSCommandPath
+$SolutionFolder = Join-Path $ProjectRoot src
+$SolutionFile = Join-Path $SolutionFolder "PInvoke.sln"
+$ToolsFolder = Join-Path $ProjectRoot tools
+$BinFolder = Join-Path $ProjectRoot "bin"
+$BinConfigFolder = Join-Path $BinFolder $Configuration
+$BinTestsFolder = Join-Path $BinConfigFolder "Tests"
+
+# Set script scope for external tool variables.
+$NuGetPath = Join-Path $ToolsFolder "NuGet.exe"
+$NuGetCommand = Get-Command $NuGetPath -ErrorAction SilentlyContinue
+$MSBuildCommand = Get-Command MSBuild.exe -ErrorAction SilentlyContinue
+$VSTestConsoleCommand = Get-Command vstest.console.exe -ErrorAction SilentlyContinue
+
+Function Get-ExternalTools {
+    if (!$NuGetCommand) {
+        Write-Host "NuGet.exe not found. Downloading to $NuGetPath..."
+        if ($PSCmdlet.ShouldProcess($sourceNuGetExe, "Download")) {
+            Invoke-WebRequest $sourceNugetExe -OutFile $NuGetPath
+        }
+        
+        $NuGetCommand = Get-Command $NuGetPath
```

Comment:
Should be 
```powershell
$script:NuGetCommand = Get-Command $NuGetPath
```

Otherwise the variable will still be empty in the script scope and nuget restore won't happen the very first time the script is executed.